src/main/java/io/neonbee/internal/cluster/entity/UnregisterEntityVerticlesHook.java

```diff
@@ -0,0 +1,93 @@
+package io.neonbee.internal.cluster.entity;
+
+import static io.neonbee.hook.HookType.CLUSTER_NODE_ID;
+import static io.vertx.core.Future.succeededFuture;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.neonbee.NeonBee;
+import io.neonbee.hook.Hook;
+import io.neonbee.hook.HookContext;
+import io.neonbee.hook.HookType;
+import io.neonbee.internal.Registry;
+import io.neonbee.internal.cluster.ClusterHelper;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+
+/**
+ * Hooks for unregistering verticle models.
+ */
+public class UnregisterEntityVerticlesHook {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UnregisterEntityVerticlesHook.class);
+
+    /**
+     * This method is called when a NeonBee instance shutdown gracefully.
+     *
+     * @param neonBee     the {@link NeonBee} instance
+     * @param hookContext the {@link HookContext}
+     * @param promise     {@link Promise} to completed the function.
+     */
+    @Hook(HookType.BEFORE_SHUTDOWN)
+    public void unregisterOnShutdown(NeonBee neonBee, HookContext hookContext, Promise<Void> promise) {
+        LOGGER.info("Unregistering models on shutdown");
+        Vertx vertx = neonBee.getVertx();
+        String nodeId = ClusterHelper.getNodeId(neonBee.getVertx());
+        unregister(vertx, nodeId).onComplete(promise)
+                .onSuccess(unused -> LOGGER.info("models unregistered successfully"))
+                .onFailure(ignoredCause -> LOGGER.error("failed to unregister models on shutdown"));
+    }
+
+    /**
+     * Unregister the entity qualified names for the node by ID.
+     *
+     * @param vertx  The Vert.x instance
+     * @param nodeId hazelcast node ID
+     * @return Future
+     */
+    public static Future<Void> unregister(Vertx vertx, String nodeId) {
```

Comment:
What do think about passing NeonBee instead of vertx + nodeId to this public method? We would need one parameter less.

Reply:
Passing NeonBee instead of vertx makes sense. However, I still need the nodeId as a parameter. See UnregisterEntityVerticlesHook#cleanup

Reply:
I understand the confusion now. It is not the NeonBee#getNodeId here. The node ID of the cluster node is needed. I renamed the corresponding parameters all to clusterNodeId to make it clearer by the name what is needed.

Reply:
Otherwise, I need to introduce a class to make it typesafe. But I did not consider it necessary.

Reply:
No that's fine! thanks for making this clear